update-center-update/uc-remediation.groovy

```diff
@@ -0,0 +1,334 @@
+    }
+
+    try {
+        debug("checking update site " + site + " with validate == " + validate);
+        FormValidation v = site.updateDirectlyNow(validate);
+        debug("form validation -> "  + v);
+        if (v.kind == FormValidation.Kind.OK) {
+            debug(site.getUrl() + " is ok");
+            setCertificateValidation(originalCheckValue);
+            return true;
+        } else if (v.kind == FormValidation.ERROR) {
+            if (v.toString().contains(_cert_error_str)) {
+                debug("cert expired error found validating " + site.getUrl());
+                setCertificateValidation(originalCheckValue);
+                return false;
+            } else {
+                debug("Some other error was found validating " + site.getUrl());
+            }
+        } else {
+            debug("Found a warning validating " + site.getUrl());
+        }
+    }
+    catch (Exception e) {
+        debug("Caught exception " + e.class + " validating cert from " + site.getUrl());
+        if (e.toString().contains(_cert_error_str)) {
+            debug("cert expired error found validating " + site.getUrl());
+            setCertificateValidation(originalCheckValue);
+            return false;
+        } else {
+            debug("Some other error was found validating " + site.getUrl());
+            setCertificateValidation(originalCheckValue);
+            return false;
+        }
+    }
+}
+
+def getDefaultOfflineUC() {
+    PersistedList <UpdateSite> sites = Jenkins.getInstance().getUpdateCenter().getSites();
+    for (UpdateSite s: sites) {
+        if (s.getUrl().equals(_offline_uc_url)) {
+            debug("Found default offline updatecenter " +s.getUrl());
+            return s;
+        }
+    }
+    debug("default offline updatecenter was not found");
+    return null;
+}
+
+def getDefaultOnlineUc() {
+    PersistedList <UpdateSite> sites = Jenkins.getInstance().getUpdateCenter().getSites();
+    for (UpdateSite s: sites) {
+        if (s.getUrl().startsWith(_online_uc_url_prefix)) {
+            debug("Found default online updatecenter " +s.getUrl());
+            return s;
+        }
+    }
+    debug("default offline updatecenter was not found");
+    return null;
+}
+
+def removeUpdateCenter(UpdateSite s) {
+    if (!_dry_run) {
+        if (Jenkins.getInstance().getUpdateCenter().getSites().remove(s)) {
+            debug.add("Removed update center " +s.getUrl());
+            return true;
+        } else {
+            debug("update center was not found, nothing removed");
+            return false;
+        }
+    } else {
+        info("dry run, not removing update center " + s.getUrl());
+        return true;
+    }
+}
+
+def debug(String msg) {
+    if (_debug) {
+        println("DEBUG: " + msg);
+    }
+}
+
+def info(String msg) {
+    println("INFO: " + msg);
+}
+'''
+// end script
+
+/**
+  * installer
+  */
+
+_init_groovy_dir = Jenkins.getInstance().getRootDir().getAbsolutePath() + "/init.groovy.d";

```

Comment:
We have some clients running their controller on Windows, so the path separator needs to be accounted for:

```suggestion
_init_groovy_dir = Jenkins.getInstance().getRootDir().getAbsolutePath() + File.separator +"init.groovy.d";
```